src/main/webapp/assets/modules/common/model/jobseekerDashboardChallenge.js

```diff
@@ -1,65 +1,84 @@
-techlooper.filter("jobseekerDashboardChallenge", function ($filter, $translate) {
+techlooper.filter("jobseekerDashboardChallenge", function ($filter, $translate, utils, jsonValue) {
   return function (input, type) {
     if (!input || input.$isRich) return input;
 
     var challenge = input;
 
     challenge.recalculate = function () {
-      var currentPhaseLowerCase = challenge.currentPhase.toLowerCase();
-      challenge.$currentPhaseLowerCase = currentPhaseLowerCase;
-      challenge.$currentPhaseTitle = $filter("translate")(currentPhaseLowerCase);
-
+      challenge.$asciiChallengeName = utils.toAscii(challenge.challengeName);
+      challenge.recalculatePhase();
       challenge.recalculateCriteria();
       challenge.recalculateAward();
       challenge.recalculateSubmissions();
-      challenge.recalculateDaysLeft();
     };
 
-    challenge.recalculateDaysLeft = function() {
+    challenge.recalculatePhase = function () {
+      var currentPhaseLowerCase = challenge.currentPhase.toLowerCase();
+      challenge.$currentPhaseLowerCase = currentPhaseLowerCase;
+      challenge.$currentPhaseTitle = $filter("translate")(currentPhaseLowerCase).toUpperCase();
+
+      challenge.$submissionMdate = moment(challenge.submissionDate, jsonValue.dateFormat);
+
+      var finishDaysLeft = utils.toNow(challenge.$submissionMdate);
+      challenge.$daysLeftTitle = finishDaysLeft > 0 ? $filter("translate")("daysLeft", {days: finishDaysLeft}) :
+        $filter("translate")("finishedOn", {date: challenge.submissionDate});
 
+      if (challenge.disqualified == true) {
+        challenge.$currentPhaseLowerCase = "disqualified";
+        challenge.$currentPhaseDescTitle = $filter("translate")("inPhase", {phase: challenge.$currentPhaseTitle});
+        challenge.$currentPhaseTitle = $filter("translate")("disqualified").toUpperCase();
+      }
+      else {
+        challenge.$currentPhaseSubmissionDate = moment(challenge.currentPhaseSubmissionDate, jsonValue.dateFormat);
+        var submitDaysLeft = utils.toNow(challenge.$currentPhaseSubmissionDate);
+        challenge.$currentPhaseDescTitle = finishDaysLeft <= 0 ? undefined : $filter("translate")("daysLeftToSubmit", {days: submitDaysLeft});
+      }
     };
 
-    challenge.recalculateSubmissions = function() {
-      _.each(challenge.submissions, function(submission) {
+    challenge.recalculateSubmissions = function () {
+      _.each(challenge.submissions, function (submission) {
         submission.$submissionURL = /^https?:\/\//.test(submission.submissionURL) ? submission.submissionURL : "http://" + submission.submissionURL;
-        submission.$phaseTitle = $filter("translate")(submission.submissionPhase.toLowerCase());
+        submission.submissionPhase && (submission.$phaseTitle = $filter("translate")(submission.submissionPhase.toLowerCase()));
```

Comment:
Line is too long.<br>Expected an assignment or function call and instead saw an expression.